dccrg.hpp

```diff
@@ -9583,6 +9583,12 @@ template <
 			unique_induced_refines.clear();
 		}
 
+		for(const auto& cell : this->cells_to_refine) {
```

Comment:
so the above loop continues until it performs a cycle after which all processes return `new_refines.size()==0`. It looks at local cells which are influenced by local or neighbour process cells. I believe this means that at the end of that looping, each process will already have information about its local cells_to_refine, and thus, deleting non-local cells in this loop should be safe.